core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -116,49 +125,57 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    */
   public static class SSHActionConfig extends PluginConfig {
     @Description("Command to be executed on the remote host. Should include filepath of script and any arguments")
+    @Macro
     private String command;
 
     @Description("Host name of the remote machine where the command needs to be executed.")
+    @Macro
     private String host;
 
     @Nullable
     @Description("Port to connect to. Defaults to 22")
+    @Macro
     private Integer port;
 
     @Description("User name used to connect to host")
+    @Macro
     private String user;
 
-    @Description("File path to Private key")
-    private String privateKeyFile;
+    @Description("Key used to lookup private key from secure storage")
+    @Macro
+    private String privateKeyLookupKey;
 
     @Nullable
-    @Description("Password associated with private key")
-    private String password;
+    @Description("Passphrase used to decrypt the provided private key associated with \"privateKeyLookupKey\"")
+    @Macro
+    private String passphrase;
 
     @Nullable
-    @Description("Variable used to hold desired key name to be associated with script output in the instance " +
-      "the script output contains information needed to be stored in the ActionContext for later use in the pipeline.")
-    private String output;
+    @Description("The key used to store the output of the command run by the action. Plugins that run at later " +
+      "stages in the pipeline can retrieve the command's output using this key through macro substitution " +
+      "${outputKey} where \"outputKey\" is the key specified.")
+    @Macro
+    private String outputKey;
```

Comment:
Since it's nullable, should a default key be given if none is provided like "sshOutput" or something?